### PR TITLE
Patch practical usability issues with Society

### PR DESCRIPTION
**Pull request by shawntabrizi**
This PR makes the following changes:

1. Currently, if you try to call `found` on the Society pallet on a running chain, it will fail since the `MaxMembers` storage item is default 0. This introduces a second parameter to the `found` extrinsic, `max_members` which will set this value to some non-zero value.

2. The Polkadot JS API does not recognize a storage item with type `Option<()>` as ever `isSome`. I change the use of `Option<()>` to `Option<bool>` for tracking suspended members. https://github.com/polkadot-js/api/issues/1724

3. This also updates the Substrate node so that Society has a genesis configuration, and thus running a new Substrate node will make society enabled and usable.

### CODE REVIEW

File: frame/society/src/lib.rs [hunk 1]
```diff
@@ -813,10 +814,12 @@ decl_module! {
 		/// Total Complexity: O(1)
 		/// # </weight>
 		#[weight = SimpleDispatchInfo::FixedNormal(10_000)]
-		fn found(origin, founder: T::AccountId) {
+		fn found(origin, founder: T::AccountId, max_members: u32) {
```
Comment (shawntabrizi) [inline frame/society/src/lib.rs]: Could also consider including a `pot` parameter which will bootstrap the pot, else we will wait 1 rotation period for it to grow.

If we include `pot`, we could also automatically transfer that amount from the founder to the Society account.

File: bin/node/cli/src/chain_spec.rs [hunk 2]
```diff
@@ -295,6 +295,11 @@ pub fn testnet_genesis(
 		}),
 		pallet_membership_Instance1: Some(Default::default()),
 		pallet_treasury: Some(Default::default()),
+		pallet_society: Some(SocietyConfig {
+			members: endowed_accounts[0..3].to_vec(),
+			pot: 100000,
```
Comment (shawntabrizi) [inline bin/node/cli/src/chain_spec.rs]: Double check this number

File: frame/society/src/lib.rs (hunk 1 above)
Comment (gavofyork) [reply frame/society/src/lib.rs]: the founder isn't meant to pay anything, and giving the `SetFounderOrigin` the capability of deducting as much as they want from any account in the system is obviously dumb.

**Comment by gavofyork** [comment]
> to `Option<bool>`

why not to a straight `bool`? it's more ergonomic and doesn't provide a dead state to the developer.

File: bin/node/cli/src/chain_spec.rs (hunk 2 above)
Comment (gavofyork) [reply bin/node/cli/src/chain_spec.rs]: pot should probably just be zero.

File: bin/node/cli/Cargo.toml [hunk 3]
```diff
@@ -72,6 +72,7 @@ pallet-transaction-payment = { version = "2.0.0", path = "../../../frame/transac
 frame-support = { version = "2.0.0", default-features = false, path = "../../../frame/support" }
 pallet-im-online = { version = "2.0.0", default-features = false, path = "../../../frame/im-online" }
 pallet-authority-discovery = { version = "2.0.0",  path = "../../../frame/authority-discovery" }
+pallet-society = { version = "2.0.0",  path = "../../../frame/society" }
```
Comment (bkchr) [inline bin/node/cli/Cargo.toml]: Why do we need this dependency?

Comment (shawntabrizi) [reply bin/node/cli/Cargo.toml]: It gets used here: https://github.com/paritytech/substrate/pull/4651/files#diff-c9f62bd741e70b8157fa630030c33a57R298

**Pull request merged**